### ISSUE TITLE
Improve relations types

### ISSUE THREAD

**Issue by Melchyore** [chirgjin/adonisjs-select-related#4]
Hi,

This package is wonderful, as it can avoid a lot of queries. However, it would be useful to have intellisense suggestions when typing relation names in `selectRelated` method.

I tried to add the types, and it worked. However, it doesn't work when you want to use nested relationships (which is obvious).

This is what I did:

```ts
interface SelectRelatedQueryMethods<
    Model extends LucidModel,
    Result = InstanceType<Model>,
    Name = ExtractModelRelations<InstanceType<Model>>
> {
    selectRelated(
        name: Name,
        options?: SelectRelatedOptions
    ): ModelQueryBuilderContract<Model, Result>
    $sideloadedRelations?: SideloadedRelations<Model>
}
```

Can we somehow split the `selectRelated` into 2 methods: `selectRelated` (for one relationship) and `selectNestedRelated` (for nested relationships)? Or find a way to not throw an error if a nested relationship is passed as parameter?

Thank you!

**Comment by chirgjin** [comment]
Hi,
I'll try to create a type to provide intellisense for nested relations as well but the below code should work in the meantime. Could you test it out & see if it works for you? If it does then you can create a PR and I'll be very happy to merge it 😁

```ts
interface SelectRelatedMethods<Model extends LucidModel, Result = InstanceType<Model>> {
        selectRelated<Name extends ExtractModelRelations<InstanceType<Model>>>(name: Name, options?: SelectRelatedOptions): ModelQueryBuilderContract<Model, Result>;
        selectRelated(name: string, options?: SelectRelatedOptions): ModelQueryBuilderContract<Model, Result>;
        $sideloadedRelations?: SideloadedRelations<Model>;
    }
 ```

**Comment by Melchyore** [comment]
Hey!

Overloading is also an option, yeah, thank you! It works, I'll create a PR for this.

**Comment by chirgjin** [comment]
Released v1.2.0
